### ISSUE TITLE
Many new false positives for Rails/HasManyOrHasOneDependent

### ISSUE THREAD

**Issue by reidab** [bbatsov/rubocop#5293]
On the current master (78d70ca), I'm seeing a ton of false positives for the Rails/HasManyOrHasOneDependent cop. I'm guessing this is related to the changes in #5235.

--------

## Expected behavior

This clearly specifies the `dependent` option
```ruby
class List < ApplicationRecord
  has_many :items, dependent: :destroy
end
```

## Actual behavior

```
Inspecting 1 file
C

Offenses:

app/models/list.rb:4:3: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
  has_many :items, dependent: :destroy
  ^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

1. run rubocop against the file above with the Rails/HasManyOrHasOneDependent cop enabled.

## RuboCop version

Installed from github master (78d70ca):
```
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by koic** [comment]
Sure. I confirmed reproduction and this issue is a regression by #5235. I'm going to open a PR to resolve this issue. Thanks for telling us issue using master branch.
